docs/production/authentication-methods.md

```diff
@@ -518,6 +518,65 @@ to the root and `engineering` subdomains:
 </saml2:Attribute>
 ```
 
+### Using Keycloak as a SAML IdP
+
+1. Make sure you reviewed [this article][saml-help-center], which
+   details how to configure Keycloak properly to use SAML with Zulip.
+2. Verify that `SOCIAL_AUTH_SAML_ENABLED_IDPS['entity_id']` and
+   `SOCIAL_AUTH_SAML_ENABLED_IDPS['url']` are correct in your Zulip
+   configuration. Specifically, if `entity_id` is
+   `https://keycloak.example.com/auth/realms/master`, then `url`
+   should be
+   `https://keycloak.example.com/auth/realms/master/protocol/saml`
+3. Your Keycloak public certificate must be saved on the Zulip server
+   as `{idp_name}.crt` in `/etc/zulip/`. You can obtain the
+   certificate from the Keycloak UI in the `Keys` tab.  Click on the
+   button `Certificate` and copy the content.
+
+   (Alternatively, open the URL in your browser
+   `https://keycloak.example.com/auth/realms/master/protocol/saml/descriptor`.
+   Replace the domain (`keycloak.example.com`) as well as the realm
+   name (`master`) in the url. The certificate is the content inside
+   `<ds:X509Certificate>[...]</ds:X509Certificate>`).
+
+   Save the certificate in an `{idp_name}.crt` file constructed as follows:
+   ```
+   -----BEGIN CERTIFICATE-----
+   {Paste the content here}
+   -----END CERTIFICATE-----
+   ```
+
+4. If you want to sign SAML requests, you have to do two things in Keycloak:
+   1. In the Keycloak client settings you setup previously, open the
+      `Settings` tab and **enable** `Client Signature Required`.
+   2. Keycloak can generate the Client private key and certificate
+      automatically, but Zulip's SAML library does not support the
+      resulting certificates.  Instead, you must generate the key and
+      certificate on the Zulip server and import them into Keycloak:
+      1. Generate **Zulip server public certificate** and the corresponding **private key**:
+         ```bash
+         openssl req -x509 -newkey rsa:2056 -keyout zulip-private-key.key \
```

Comment:
Ah, this remind of the mystery of `2056` value from https://github.com/zulip/zulip/pull/18648#discussion_r643414778 - not sure what should be done here :confounded:

Reply:
I'd assume 2056 is correct given that context.